src/main.ts

```diff
@@ -1,49 +1,129 @@
+            const recommendation = reviewElement.find('.c-post__recommendation').text().trim();
+            const rating = reviewElement.find('.c-rating-widget').attr('data-rating');
+            const summary = reviewElement.find('p.c-post__summary').text().trim();
+
+            const pros: string[] = [];
+
+            const prosContainer = reviewElement.find(
+                'ul.c-attributes-list.c-attributes-list--pros.c-attributes-list--circle.o-block-list.o-block-list--snug',
+            );
+
+            prosContainer.find('li.c-attributes-list__item').each((_idx, el) => {
+                pros.push($(el).text().trim());
+            });
 
-        // Extract title from the page.
-        const title = $('title').text();
-        log.info(`${title}`, { url: request.loadedUrl });
+            const cons: string[] = [];
 
-        // Save url and title to Dataset - a table-like storage.
-        await Dataset.pushData({ url: request.loadedUrl, title });
+            const consContainer = reviewElement.find(
+                'ul.c-attributes-list.c-attributes-list--cons.c-attributes-list--circle.o-block-list.o-block-list--snug',
+            );
+
+            consContainer.find('li.c-attributes-list__item').each((_idx, el) => {
+                cons.push($(el).text().trim());
+            });
+
+            const shopReply = reviewElement.find('.c-post-response > p').text().trim();
+
+            const pageReview: Review = {
+                shopName,
+                author,
+                reviewAt,
+                recommendation,
+                rating,
+                pros,
+                cons,
+                summary,
+                shopReply,
+            };
+
+            pageReviews.push(pageReview);
+        });
+
+        const reviewsNeeded = maxShopReviews - (state.scrapedReviewsPerShop[shopName] || 0);
+        pageReviews = pageReviews.slice(0, reviewsNeeded);
+
+        if (pageReviews.length > 0) {
+            // update state must be before pushData to prevent race condition
+            state.scrapedReviewsPerShop[shopName] = (state.scrapedReviewsPerShop[shopName] || 0) + pageReviews.length;
+
+            await Dataset.pushData(pageReviews);
+
+            log.info(
+                `Successfully saved ${state.scrapedReviewsPerShop[shopName]}/${maxShopReviews} reviews from page ${request.url}`,
+            );
+        } else {
+            log.info(`No reviews found on page ${request.url}`);
+        }
+
+        // stop processing new pages when reviews limit is reached
+        if (state.scrapedReviewsPerShop[shopName] < maxShopReviews) {
+            await enqueueLinks({
+                strategy: EnqueueStrategy.SameDomain,
+                selector: 'a.c-pagination__controls[rel="next"]',
+                regexps: [/\?f=\d+#filtr/],
+            });
+        }
     },
 });
 
-await crawler.run(startUrls);
+await crawler.run(shopUrls);
+
+Actor.on('migrating', async () => {
+    await Actor.reboot();
+});
```

Comment:
This needs to be registered somewhere at the start, at worst before crawler.run otherwise it will not trigger during crawling.

Reply:
Good catch! Fixed.